### DISCUSSION TITLE
How to convert path to valid underlying OS representation ?

### DISCUSSION THREAD

**Discussion by MichelMunoz** [nushell/nushell#16662] [Q&A]
### Question

(simplified) **Use case** : I have to call an external os-dependent command only known at runtime using something like :
 `nu -C $"^somecommand \"($mypath)\""` (path with quotes).

The path comes from a _path_ variable, say `$mypath`
- on linux mypath=`"/some/path"`
- on window mypath=`"c:\\some\\path"`

**Problem:** implicit path-to-string renders `"c:\\some\\path"` as `"c:\some\path"` in the command text (executed by `nu -c`) on windows which is invalid in the OS. ("\p" is seen as invalid escape caracter)

**Question** surprisingly, according to the docs there seems to be no way to render a path object to underlying-OS valid representation. (ex. escape spaces on linux, escape backslashes on windows, etc.) which is a recurring problem for shell work.
Is there some command that I missed somewhere ? (like `to os-path` or `path to-string` or something)

**Comment by razcore-rad** [comment]
I'm not sure if this is related, but I ran into an issue with external-path strings as well. The other day I was setting up `wine` to install some VST plugins with `yabridgectl` and I just couldn't get the proper string format for paths with spaces. I have no idea if I don't understand something or there isn't a way to properly escape paths for external commands.

All of these failed:
```
yabridgectl add ~/.wine/drive_c/Program\ Files/VstPlugins
yabridgectl add (echo "~/.wine/drive_c/Program\\ Files/VstPlugins" | to text)
yabridgectl add (echo `~/.wine/drive_c/Program\ Files/VstPlugins` | to text)
yabridgectl add "~/.wine/drive_c/Program\\ Files/VstPlugins"
yabridgectl add "~/.wine/drive_c/Program\ Files/VstPlugins"
yabridgectl add "~/.wine/drive_c/Program Files/VstPlugins"
yabridgectl add '~/.wine/drive_c/Program Files/VstPlugins'
yabridgectl add '~/.wine/drive_c/Program\ Files/VstPlugins'
yabridgectl add ~/.wine/drive_c/Program\ Files/VstPlugins
```

For example:
```
❯ yabridgectl add ~/.wine/drive_c/Program\ Files/VstPlugins
error: Found argument 'Files/VstPlugins' which wasn't expected, or isn't valid in this context

USAGE:
    yabridgectl add <path>

For more information try --help
```
